internal/controller/kyma_controller.go

```diff
@@ -405,21 +406,73 @@ func (r *KymaReconciler) handleDeletingState(ctx context.Context, kyma *v1beta2.
 		}
 
 		r.RemoteClientCache.Del(client.ObjectKeyFromObject(kyma))
-		if err := remote.RemoveFinalizerFromRemoteKyma(ctx, r.RemoteSyncNamespace); client.IgnoreNotFound(err) != nil {
+		if err := remote.RemoveFinalizersFromRemoteKyma(ctx, r.RemoteSyncNamespace); client.IgnoreNotFound(err) != nil {
 			err = fmt.Errorf("error while trying to remove finalizer from remote: %w", err)
 			r.enqueueWarningEvent(kyma, deletionError, err)
 			return ctrl.Result{}, err
 		}
 
-		logger.Info("removed remote finalizer")
+		logger.Info("removed remote finalizers")
+	}
+
+	manifestsExist, err := r.cleanupManifestCRs(ctx, kyma)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if manifestsExist {
+		return ctrl.Result{Requeue: true}, nil
 	}
 
 	r.Metrics.CleanupMetrics(kyma.Name)
 
 	controllerutil.RemoveFinalizer(kyma, shared.KymaFinalizer)
+
 	return ctrl.Result{Requeue: true}, r.updateKyma(ctx, kyma)
 }
 
+func (r *KymaReconciler) cleanupManifestCRs(ctx context.Context, kyma *v1beta2.Kyma) (bool, error) {
+	relatedManifests, err := r.getRelatedManifestCRs(ctx, kyma)
+	if err != nil {
+		err = fmt.Errorf("error while trying to get manifests: %w", err)
+		r.enqueueWarningEvent(kyma, deletionError, err)
+		return false, err
+	}
+
+	if !r.relatedManifestCRsAreDeleted(relatedManifests) {
```

Comment:
if you check positive `if r.relatedManifestCRsAreDeleted(relatedManifests)` will make this code much clean.

```
if r.relatedManifestCRsAreDeleted(relatedManifests) {
  return false, nil
}
if err = r.deleteManifests(ctx, relatedManifests); err != nil {
	r.enqueueWarningEvent(kyma, deletionError, err)
	return true, err
}
return true, nil

```